### ISSUE TITLE
Failed to support LinkedIn login

### ISSUE THREAD

**Issue by fogfish** [auth0/auth0-flutter#432]
### Checklist

- [X] The issue can be reproduced in the [auth0_flutter sample app](https://github.com/auth0-samples/auth0-flutter-samples/tree/main/sample) (or N/A).
- [X] I have looked into the [Readme](https://github.com/auth0/auth0-flutter/tree/main/auth0_flutter#readme), [Examples](https://github.com/auth0/auth0-flutter/blob/main/auth0_flutter/EXAMPLES.md), and [FAQ](https://github.com/auth0/auth0-flutter/blob/main/auth0_flutter/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://pub.dev/documentation/auth0_flutter/latest/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/auth0-flutter/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

Usage of LinkedIn login and flutter web fails with the following error:

```
Error: TypeError: null: type 'Null' is not a subtype of type 'String'
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 297:3   throw_
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1385:3                          _failedAsCheck
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1363:3                          _generalAsCheckImplementation
dart-sdk/lib/_internal/js_shared/lib/js_util_patch.dart 81:5                  getProperty
packages/auth0_flutter/src/web/extensions/web_exception_extensions.dart 6:19  WebExceptionExtension.fromJsObject
packages/auth0_flutter/src/web/auth0_flutter_plugin_real.dart 128:49          credentials
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50            <fn>
```

The root cause is 
https://github.com/auth0/auth0-flutter/blob/main/auth0_flutter/lib/src/web/extensions/user_profile_extension.dart#L53

The lib expect locale claim to be String. However, LinkedIn return nested JSON Object: 
```
"locale": {
    "country": "US",
    "language": "en"
  },
``` 

### Reproduction

1. Config LinkedIn OAuth app
2. Config Auth0 to using this app
3. Run [auth0_flutter sample app](https://github.com/auth0-samples/auth0-flutter-samples/tree/main/sample) with html renderer
4. Login to LinkedIn

### auth0_flutter version

1.6.0

### Flutter version

Flutter 3.19.3 • channel stable

### Platform

Web

**Comment by extphuphan** [comment]
Hi @fogfish . Do you have any solution for that?

**Comment by fogfish** [comment]
@extphuphan yes! before auth0 would make a fix to this issue. You have to patch it locally.

```
  auth0_flutter: # ^1.6.0
    path: ./vendor/auth0-flutter/auth0_flutter
```

```
--- a/auth0_flutter/lib/src/web/extensions/user_profile_extension.dart
+++ b/auth0_flutter/lib/src/web/extensions/user_profile_extension.dart
@@ -50,7 +50,7 @@ extension UserProfileExtension on UserProfile {
       gender: claims[PublicClaims.gender] as String?,
       birthdate: claims[PublicClaims.birthdate] as String?,
       zoneinfo: claims[PublicClaims.zoneinfo] as String?,
-      locale: claims[PublicClaims.locale] as String?,
+//      locale: claims[PublicClaims.locale] as String?,
       phoneNumber: claims[PublicClaims.phoneNumber] as String?,
       isPhoneNumberVerified: claims[PublicClaims.phoneNumberVerified] as bool?,
       address: address,
```
